modules/gapi/src/api/gorigin.cpp

```diff
@@ -21,8 +21,15 @@ cv::GOrigin::GOrigin(GShape s,
 }
 
 cv::GOrigin::GOrigin(GShape s, cv::gimpl::ConstVal v)
-    : shape(s), node(cv::GNode::Const()), value(v), port(INVALID_PORT), kind(cv::detail::OpaqueKind::CV_UNKNOWN)
+    : shape(s), node(cv::GNode::Const()), value(v), port(INVALID_PORT)
 {
+    switch (v.index())
+    {
+    case cv::gimpl::ConstVal::index_of<detail::VectorRef>():
+        kind = util::get<detail::VectorRef>(v).getKind();
+    default:
+        kind = cv::detail::OpaqueKind::CV_UNKNOWN;
+    }
```

Comment:
```suggestion
kind = holds_alternative<VectorRef>(v)
    ? util::get<detail::VectorRef>(v).getKind()
    : cv::detail::OpaqueKind::CV_UNKNOWN;
```

Reply:
Ok. But, what if we want to add initialization for gopaque?

Reply:
Fixed.

Reply:
Good point btw, extending it to opaque will really look better with `switch()`. You could leave a placeholder here before `default` to indicate that `OpaqueRef` will be the next case - then it'd be more clear